bahmnicore-api/src/test/java/org/bahmni/module/bahmnicore/dao/impl/BahmniPatientDaoImplLuceneIT.java

```diff
@@ -213,4 +212,43 @@ public void shouldNotReturnDuplicatePatientsEvenIfTwoIdentifiersMatches() {
         assertTrue(patient.getExtraIdentifiers().contains("200006"));
     }
 
+    @Test
+    public void shouldSearchSimilarPatientByPatientName() {
+        List<PatientResponse> patients = patientDao.getSimilarPatientsUsingLuceneSearch("Peet", "", "c36006e5-9fbb-4f20-866b-0ece245615a1", 5);
+        PatientResponse patient1 = patients.get(0);
+        PatientResponse patient2 = patients.get(1);
+
+        assertEquals(2, patients.size());
+        assertEquals(patient1.getGivenName(), "Horatio");
+        assertEquals(patient1.getMiddleName(), "Peeter");
+        assertEquals(patient1.getFamilyName(), "Sinha");
+        assertEquals(patient2.getGivenName(), "John");
+        assertEquals(patient2.getMiddleName(), "Peeter");
+        assertEquals(patient2.getFamilyName(), "Sinha");
+    }
+
+    @Test
+    public void shouldSearchSimilarPatientByPatientNameAndUseLimitResult() {
+        List<PatientResponse> patients = patientDao.getSimilarPatientsUsingLuceneSearch("Peet", "", "c36006e5-9fbb-4f20-866b-0ece245615a1", 1);
+        assertEquals("Should limit number of results",1, patients.size());
+        PatientResponse patient1 = patients.get(0);
+
+        assertEquals(patient1.getGivenName(), "Horatio");
+        assertEquals(patient1.getMiddleName(), "Peeter");
+        assertEquals(patient1.getFamilyName(), "Sinha");
+    }
+
+    @Test
+    public void shouldSearchSimilarPatientByPatientNameAndGender() {
+        List<PatientResponse> patients = patientDao.getSimilarPatientsUsingLuceneSearch("Peet", "F", "c36006e5-9fbb-4f20-866b-0ece245615a1", 5);
+        PatientResponse patient1 = patients.get(0);
+        
+        assertEquals(1, patients.size());
+        assertEquals(patient1.getGivenName(), "John");
+        assertEquals(patient1.getMiddleName(), "Peeter");
+        assertEquals(patient1.getFamilyName(), "Sinha");
+    }
+
+    //TODO missing tests: by limit, parameters verification
```

Comment:
remove the TODO/replace it with missing tests.
you tested limit in `shouldSearchSimilarPatientByPatientNameAndUseLimitResult`

what do you want to test in `parameters verification` ?